src/vanilla.jl

```diff
@@ -16,60 +23,51 @@ mutable struct ValueIterationPolicy <: Policy
     action_map::Vector # Maps the action index to the concrete action type
     include_Q::Bool # Flag for including the Q-matrix
     mdp::Union{MDP,POMDP} # uses the model for indexing in the action function
-    # constructor with an optinal initial value function argument
-    function ValueIterationPolicy(mdp::Union{MDP,POMDP};
-                                  utility::Vector{Float64}=Array{Float64}(0),
-                                  include_Q::Bool=true)
-        ns = n_states(mdp)
-        na = n_actions(mdp)
-        self = new()
-        if !isempty(utility)
-            @assert first(size(utility)) == ns "Input utility dimension mismatch"
-            self.util = utility
-        else
-            self.util = zeros(ns)
-        end
-        self.action_map = ordered_actions(mdp)
-        self.policy = zeros(Int64,ns)
-        include_Q ? self.qmat = zeros(ns,na) : self.qmat = zeros(0,0)
-        self.include_Q = include_Q
-        self.mdp = mdp
-        return self
-    end
-    # constructor for solved q, util and policy
-    function ValueIterationPolicy(mdp::Union{MDP,POMDP}, q::Matrix{Float64}, util::Vector{Float64}, policy::Vector{Int64})
-        self = new()
-        self.qmat = q
-        self.util = util
-        self.policy = policy
-        self.action_map = ordered_actions(mdp)
-        self.include_Q = true
-        self.mdp = mdp
-        return self
+end
+
+# constructor with an optinal initial value function argument
+function ValueIterationPolicy(mdp::Union{MDP,POMDP};
+                              utility::Vector{Float64}=Vector{Float64}(0),
```

Comment:
Why not:

```julia
utility::Vector{Float64}=zeros(n_states(mdp)),
policy::Vector{Int64}=zeros(Int64, n_states(mdp)),
```

Then you won't need the extra if else.

Reply:
Good point, I made the change.